README.md

```diff
@@ -96,6 +96,7 @@ Emulated Vim plugins:
 
 * vim-surround
 * vim-multiple-cursors
+* argtextobj.vim
```

Comment:
Could you please remove `argtextobj` from readme? We will add this information after the public release with this functionality.

Reply:
@AlexPl292, thank you very much for the review!

> I can see A LOT of pretty complicated logic here. I'm not sure I can understand it completely and such code requires good tests.

The extension uses a heuristic algorithm and thus has a somewhat limited application given the huge syntax space of multiple languages. An alternative would be to use the a proper language support framework, but it would limit the usefulness of the extension. In particular it can work inside comments and incomplete source code. In addition, if the heuristic fails, you'd simply undo it's effect. While I definitely agree the existing level of testing was subpar, I'm not sure how valuable or feasible it would be to have an exhaustive set of tests -- I've been using it daily for a couple of years now without issues. Nevertheless, I've added a couple of corner-case tests (and fixed a few bugs ;-) and got to 100% coverage:
![Screenshot_20190710_233923](https://user-images.githubusercontent.com/1772013/60974833-22f93c00-a36e-11e9-9e57-f83d9f40bd59.png)
Please let me know if you would like to test any specific cases -- happy to add them.

I completely understand if you decide that the new level of testing is still not acceptable. In this I would like to know if it would be technically feasible to release the extension as a separate plugin.

> After some playing with this extension, I got an exception while executing `daa` on empty line

Fixed.

>  and I've cleared a body of a function with `cia`.

As I mentioned above, the algorithm is heuristic and it doesn't distinguish between a function body and a brace-initialization in C++. I did put a mitigation for this in the form of limiting the search to 10 lines or 800 characters from the caret position.

> Tests with good coverage are necessary for this PR. And there are a lot of test cases that are out of your existing test: empty file, empty line, executing not on argument, etc. Negative testing should also be provided (when syntax is broken like `function (int x \n{ ... }`, `function (<int x [\n{ ... }`).